plugins/modules/keycloak_identity_provider.py

```diff
@@ -542,10 +542,14 @@ def main():
                     old_mapper = dict()
             new_mapper = old_mapper.copy()
             new_mapper.update(change)
```

Comment:
This line is problematic, it overrides all fields that weren't provided in the playbook with `None`. This wasn't introduced by this PR, but was already a problem with this module.

Maybe fixing this already improves the situation (without the other changes)?

Reply:
Shouldn't `dict.update()` modify only the keys which are present in the argument? I.e. override only the fields which were provided in the playbook.

The only problem we have observed with this were caused by the fact that `dict.update` is not recursive, i.e. for nested lists of dictionaries it did not work properly. That is what should be fixed by the proposed PR, alas only for the `mappers` attribute and not on general level.

This approach to apply updates is used throughout all of the keycloak modules (perhaps the same author?).

Reply:
Ah wait, I obviously missed the `change = dict((k, v) for k, v in change.items() if change[k] is not None)` line. So yes, `dict.update()` is fine here. Will have to look at this again.